lib/publisher/auditpublisher.go

```diff
@@ -0,0 +1,105 @@
+package publisher
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/cloudflare/cfssl/log"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/events"
+	"github.com/gravitational/trace"
+)
+
+const (
+	//default sqs queue name
+	defaultAuditSQSQueueName = "remote-access-audit-event-notification-queue"
+)
+
+var (
+	eventsToPublish = getSupportedAuditTypes()
+)
+
+type AuditPublisher interface {
+	apievents.Emitter
+}
+
+type AuditPublisherConfig struct {
+	Enabled      bool
+	SQSQueueName string
+}
+
+type AuditPublisherService struct {
+	publisher AuditMessagePublisher
+}
+
+type AuditMessage struct {
+	MessageBody string
+}
+
+// AuditMessagePublisher responsible for publish the audit message to destination
+type AuditMessagePublisher interface {
+	Publish(audit AuditMessage) error
+}
+
+func (cfg *AuditPublisherConfig) CheckAndSetDefaults() error {
+	if cfg.SQSQueueName == "" {
+		cfg.SQSQueueName = defaultAuditSQSQueueName
+	}
+	return nil
+}
+
+func NewAuditPublisher(cfg AuditPublisherConfig) AuditPublisher {
+	if cfg.Enabled {
+		log.Info("Audit publishing is eanbled")
+		cfg.CheckAndSetDefaults()
+		return &AuditPublisherService{
+			publisher: NewAuditMessagePublisherService(cfg),
+		}
+	}
+	log.Info("Audit publishing is disabled")
+	return &noOpAuditPublisher{}
+}
+
+// EmitAuditEvent implements AuditPublisher
+func (s *AuditPublisherService) EmitAuditEvent(ctx context.Context, event apievents.AuditEvent) error {
+	//filter the required event to publish
+	if _, ok := eventsToPublish[event.GetType()]; !ok {
+		log.Infof("Skipping the audit event id:%v type: %v", event.GetID(), event.GetType())
+		return nil
+	}
+
+	log.Infof("Publishing the audit event [ id %v  cluster: %v and type: %v ]", event.GetID(), event.GetClusterName(), event.GetType())
+	messageBody, err := json.Marshal(event)
+	if err != nil {
+		log.Errorf("Failed to serialize the audit event to json err: %v", err)
+		return trace.Wrap(err)
+	}
+
+	err = s.publisher.Publish(AuditMessage{
+		MessageBody: string(messageBody),
+	})
+
+	if err != nil {
+		log.Infof("Failed to publish the audit event [ id %v  cluster: %v and type: %v ]", event.GetID(), event.GetClusterName(), event.GetType())
+		return err
+	}
+
+	log.Infof("Published the audit event [ id %v  cluster: %v and type: %v ]", event.GetID(), event.GetClusterName(), event.GetType())
+	return nil
+}
+
+func getSupportedAuditTypes() map[string]string {
+	auditTypes := make(map[string]string)
+	auditTypes[events.SessionStartEvent] = events.SessionStartEvent
+	auditTypes[events.SessionEndEvent] = events.SessionEndEvent
+	auditTypes[events.AppSessionStartEvent] = events.AppSessionStartEvent
+	return auditTypes
+}
```

Comment:
Please change this to configuration, so that we dont have to change code to start supporting new events.